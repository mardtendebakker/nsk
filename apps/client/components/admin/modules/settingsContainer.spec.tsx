/* eslint-disable react/prop-types */
import React from 'react';
import { render, screen } from '@testing-library/react';
import SettingsContainer from './settingsContainer';

jest.mock('@mui/material', () => ({
  Box: ({ children }) => <div data-testid="box">{children}</div>,
  Card: ({ children }) => <div data-testid="card">{children}</div>,
  CardContent: ({ children }) => <div data-testid="card-content">{children}</div>,
}));

jest.mock('./menu', () => ({
  __esModule: true,
  default: () => <div data-testid="menu">Menu Component</div>,
}));

describe('SettingsContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    function TestChild() {
      return <div>Test Child Content</div>;
    }
    const { asFragment } = render(<SettingsContainer><TestChild /></SettingsContainer>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the container with all components', () => {
    function TestChild() {
      return <div>Test Child Content</div>;
    }
    render(<SettingsContainer><TestChild /></SettingsContainer>);

    expect(screen.getByTestId('card')).toBeInTheDocument();
    expect(screen.getByTestId('card-content')).toBeInTheDocument();
    expect(screen.getAllByTestId('box')).toHaveLength(2);
    expect(screen.getByTestId('menu')).toBeInTheDocument();
  });

  it('renders children correctly', () => {
    function TestChild() {
      return <div>Test Child Content</div>;
    }
    render(<SettingsContainer><TestChild /></SettingsContainer>);

    expect(screen.getByText('Test Child Content')).toBeInTheDocument();
  });

  it('renders the menu component', () => {
    function TestChild() {
      return <div>Test Child Content</div>;
    }
    render(<SettingsContainer><TestChild /></SettingsContainer>);

    expect(screen.getByText('Menu Component')).toBeInTheDocument();
  });
});
