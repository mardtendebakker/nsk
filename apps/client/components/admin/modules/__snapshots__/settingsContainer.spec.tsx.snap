// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsContainer matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="card"
  >
    <div
      data-testid="card-content"
    >
      <div
        data-testid="box"
      >
        <div
          data-testid="menu"
        >
          Menu Component
        </div>
      </div>
      <div
        data-testid="box"
      >
        <div>
          Test Child Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
